clients/client-java/src/main/java/com/datastrato/gravitino/client/ErrorHandlers.java

```diff
@@ -350,21 +355,20 @@ public ErrorResponse parseResponse(int code, String json, ObjectMapper mapper) {
      */
     @Override
     public void accept(ErrorResponse errorResponse) {
-      if (errorResponse.getType() != null) {
-        switch (errorResponse.getType()) {
-          case OAuth2ClientUtil.INVALID_CLIENT_ERROR:
-            throw new UnauthorizedException(
-                "Not authorized: %s: %s", errorResponse.getType(), errorResponse.getMessage());
-          case OAuth2ClientUtil.INVALID_REQUEST_ERROR:
-          case OAuth2ClientUtil.INVALID_GRANT_ERROR:
-          case OAuth2ClientUtil.UNAUTHORIZED_CLIENT_ERROR:
-          case OAuth2ClientUtil.UNSUPPORTED_GRANT_TYPE_ERROR:
-          case OAuth2ClientUtil.INVALID_SCOPE_ERROR:
-            throw new BadRequestException(
-                "Malformed request: %s: %s", errorResponse.getType(), errorResponse.getMessage());
-        }
+      switch (errorResponse.getType()) {
+        case OAuth2ClientUtil.INVALID_CLIENT_ERROR:
+          throw new UnauthorizedException(
+              "Not authorized: %s: %s", errorResponse.getType(), errorResponse.getMessage());
+        case OAuth2ClientUtil.INVALID_REQUEST_ERROR:
+        case OAuth2ClientUtil.INVALID_GRANT_ERROR:
+        case OAuth2ClientUtil.UNAUTHORIZED_CLIENT_ERROR:
+        case OAuth2ClientUtil.UNSUPPORTED_GRANT_TYPE_ERROR:
+        case OAuth2ClientUtil.INVALID_SCOPE_ERROR:
+          throw new BadRequestException(
+              "Malformed request: %s: %s", errorResponse.getType(), errorResponse.getMessage());
+        default:
+          super.accept(errorResponse);
```

Comment:
If errorResponse.getType() == null, `super.accept(errorResponse)` is called in the default clause not outside switch statement.